### PR TITLE
Devin fix

### PR DESCRIPTION
**Pull request by Dsenza**
Potential fix for heroku

### CODE REVIEW

File: server/boot/01-frontend-server.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+var Server = require('../frontend-server.js');
+var app = Server.app();
+
+
+// If the node_env is NOT set to production, run the webpackdev server
+// Uses the webpack.dev.config.js file for webpack configuration.
+if (process.env.NODE_ENV !== 'production') {
+
+	var webpack = require('webpack');
```
Comment (wdoug) [inline server/boot/01-frontend-server.js]: We'll need this required outside of this if statement

Comment (Dsenza) [inline server/boot/01-frontend-server.js]: Why does the webpack require have to be outside the if statement? It's only used for the dev-server? In production, it's just running a simple express app.

File: server/datasources.production.js [hunk 2]
```diff
@@ -1,16 +1,16 @@
...
-    port: process.env.PG_PORT,
-    database: process.env.PG_DB,
-    username: process.env.PG_USER,
-    password: process.env.PG_PW,
-    name: 'postgres',
-    connector: 'postgresql'
+  "postgres": {
+    "host": "localhost",
+    "port": 5432,
+    "database": "devin",
```
Comment (wdoug) [inline server/datasources.production.js]: Definitely don't want to add this in the pull request!

File: server/boot/01-frontend-server.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	  if (err) {
+	    return console.log(err);
+	  }
+
+	  console.log('Frontend located at http://localhost:8080/client/');
+	});
+} 
+// Otherwise, run a simple express server to serve up the frontend files.
+else {
+	app.listen(8080)
```
Comment (wdoug) [inline server/boot/01-frontend-server.js]: We don't need to do this. The app will already be listening on port 3000 (or whatever heroku runs it on).

We do, however, need to do a one-time webpack (see https://github.com/codefordenver/ketohero/blob/combine-api-and-devin/server/boot/01_webpack.js) for what I had started.

File: server/frontend-server.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+var express = require('express')
+
+module.exports = {
+	app: function() {
+		var app = express();
+		var indexPath = path.join(__dirname, '../dist/index.html');
+		var publicPath = express.static(path.join(__dirname, '../dist/public'));
+
+		app.use('/public', publicPath);
+		app.get('/', function (_, res) { res.sendFile(indexPath) });
```
Comment (wdoug) [inline server/frontend-server.js]: I'd probably still define the first param as `req` or `request` even though we aren't using it here.

Comment (wdoug) [inline server/frontend-server.js]: Also, we'll want to serve this for any paths (that don't start with `/api/`) so the router works for hard reloads.

File: dist/index.html [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<!doctype html>
```
Comment (wdoug) [inline dist/index.html]: We already have a index file in `client` can we use that instead?

File: package.json [hunk 6]
```diff
@@ -3,37 +3,39 @@
...
-    "redux": "^3.3.1",
+    "babel-core": "^6.7.4",
+    "babel-eslint": "^6.0.0",
+    "babel-loader": "^6.2.4",
+    "babel-preset-es2015": "^6.6.0",
+    "babel-preset-react": "^6.5.0",
+    "babel-preset-stage-0": "^6.5.0",
     "compression": "^1.0.3",
     "cors": "^2.5.2",
+    "express": "^4.13.4",
```
Comment (wdoug) [inline package.json]: I don't think this is used.

Comment (Dsenza) [inline package.json]: I was, until I realized loopback is built on it, I need to remove it again.

File: .gitignore [hunk 7]
```diff
@@ -19,4 +19,4 @@ node_modules
 npm-debug.log
 server/datasources.local.json
 .vagrant/
-dist/public
+client/dist
```
Comment (wdoug) [inline .gitignore:22]: 👍

**Pull request merged**